Controller/ControllerTrait.php

```diff
@@ -83,6 +89,30 @@ protected function routeRedirectView($route, array $parameters = [], $statusCode
      */
     protected function handleView(View $view)
     {
-        return $this->get('fos_rest.view_handler')->handle($view);
+        return $this->getViewHandler()->handle($view);
+    }
+
+    /**
+     * Get the ViewHandler.
+     *
+     * @return ViewHandlerInterface
+     */
+    protected function getViewHandler()
+    {
+        if (!$this->viewhandler instanceof ViewHandlerInterface) {
+            $this->viewhandler = $this->get('fos_rest.view_handler');
+        }
+
+        return $this->viewhandler;
+    }
+
+    /**
+     * Get the ViewHandler.
+     *
+     * @@param View $view
+     */
+    protected function setViewHandler(ViewHandlerInterface $viewhandler)
```

Comment:
because i wanted this trait also to be useful when not using the ContainerAwareTrait. will send a PR in a few mins that splits that use case into two separate traits